src/Board.py

```diff
@@ -154,15 +158,15 @@ def noMatingMaterial(self):
             return True
         return False
 
-    def getLastMove(self):
+    def getLastMove(self) -> Move:  # type: ignore[return]  # TODO: add consistent return for else condition
         if self.history:
-            return self.history[-1][0]
+            return self.history[-1][0]  # type: ignore[no-any-return]
 
-    def getLastPieceMoved(self):
+    def getLastPieceMoved(self) -> Piece:  # type: ignore[return]  # TODO: add consistent return for else condition
         if self.history:
-            return self.history[-1][0].piece
+            return self.history[-1][0].piece  # type: ignore[no-any-return]
```

Comment:
Consistent return value seems to be `None` in this case. Or do you have something else in mind?

Reply:
Hmm, not really, I think. See that there is a return statement within `if` condition, and if that `if` is satisfied, a `Piece` type is returned. But if the `if` is not executed then the function returns `None`, as there's no `else` condition here.

See Mypy when this is changed to 
```
    def getLastPieceMoved(self) -> Optional[Piece]:
        if self.history:
            return self.history[-1][0].piece
```

```
~/Projects/command-line-chess (add-mypy ✗) mypy --show-error-codes --pretty --strict .
src/Board.py:166: error: Missing return statement  [return]
        def getLastPieceMoved(self) -> Optional[Piece]:
        ^
src/Board.py:168: error: Returning Any from function declared to return "Optional[Piece]"  [no-any-return]
                return self.history[-1][0].piece
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 2 errors in 1 file (checked 16 source files)
```

And without `Optional` in return type:
```
~/Projects/command-line-chess (add-mypy ✗) mypy --show-error-codes --pretty --strict .
src/Board.py:166: error: Missing return statement  [return]
        def getLastPieceMoved(self) -> Piece:
        ^
src/Board.py:168: error: Returning Any from function declared to return "Piece"  [no-any-return]
                return self.history[-1][0].piece
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 2 errors in 1 file (checked 16 source files)
```

Reply:
`history` is declared to be `list[list]` which is automatically translated to `list[list[Any]]`.
So I assume mypy isn't happy for the following reasons:
- Line 166: mypy can't infer the return type because the element which is accessed (`history[-1][0]` where `[0]` is the important part) is declared to be `Any`
- Line 171: same as in line 166. mypy can't infer what type `history[-1][0].piece` is because `history[-1][0]` is already `Any`.

Reply:
You're right, I missed that `history[-1][1]` is not always `None`. Should be OK now.